### PR TITLE
Add GEOMEAN formula

### PR DESCRIPTION
**Pull request by iamartyom**
This pr contains implementation GEOMEAN formula with test cases.

### CODE REVIEW

File: ClosedXML/ClosedXML.csproj [hunk 1]
```diff
@@ -15,12 +15,13 @@
     <Copyright>MIT</Copyright>
     <PackageProjectUrl>https://github.com/ClosedXML/ClosedXML</PackageProjectUrl>
     <PublishRepositoryUrl>true</PublishRepositoryUrl>
-    <PackageIconUrl>https://raw.githubusercontent.com/ClosedXML/ClosedXML/develop/resources/logo/nuget-logo.png</PackageIconUrl>
+    <PackageIconUrl></PackageIconUrl>
```
Comment (igitur) [inline ClosedXML/ClosedXML.csproj:18]: Please rather do this in a separate PR.

Comment (iamartyom) [reply ClosedXML/ClosedXML.csproj:18]: Sure.
I created new pr with this fix - https://github.com/ClosedXML/ClosedXML/pull/1516.

**Comment by igitur** [comment]
Thanks for the contribution.
